### DISCUSSION TITLE
Microprocessors

### DISCUSSION THREAD

**Discussion by aaraujo666** [hneemann/Digital#1241] [Ideas]
How hard would it be to implement microprocessor components into the application? I'm thinking along the lines of 6502, Z80, 6809, 6803, 68000, etc. 

I downloaded the source code, tried to implement a 6502 processor (just two instructions to see if it worked), but ran into an issue that when I "mvn package" I get failed tests in code that is not part of my implementation. Not sure if it is a toolset issue, or something else. 

I attached my code to this discusssion, if you want to take a look se and see if I am on the right track, I will continue and submit a PR at some point. If you have any insight into my test failures, that would be appreciated as well.

[cpu.zip](https://github.com/hneemann/Digital/files/13771777/cpu.zip)

**Comment by innot** [comment]
Hi,

I have just released a [custom Digital Plugin](https://github.com/innot/PETComponentsPlugin) that includes a cycle accurate [6502 simulation](https://github.com/innot/Sim6502Java) written in Java.

While I do not indent to write simulations for other processors, maybe this plugin can be used as a starting step to simulate other CPUs or special chips that would be to complex to implement with Digital's logic elements.

brgds,
Thomas
